### ISSUE TITLE
ComponentBase: TypeError: Cannot read properties of undefined (reading 'unstyled')

### ISSUE THREAD

**Issue by mondaychen** [primefaces/primereact#5168]
### Describe the bug

I'm trying to upgrade my app to latest version of primereact (10.0.7) when I hit this error when the page loads

```
Unhandled Runtime Error
TypeError: Cannot read properties of undefined (reading 'unstyled')

Call Stack
isUnstyled
../node_modules/primereact/componentbase/componentbase.esm.js (177:0)
cx
../node_modules/primereact/componentbase/componentbase.esm.js (182:0)
eval
../node_modules/primereact/ripple/ripple.esm.js (161:25)
renderWithHooks
../node_modules/react-dom/cjs/react-dom.development.js (16305:0)
```

I toggled on the "pause exceptions" option on Chrome DevTools and got this:
<img width="648" alt="image" src="https://github.com/primefaces/primereact/assets/1001890/ead3302e-237b-4efd-8eca-c609be36c89d">
<img width="654" alt="image" src="https://github.com/primefaces/primereact/assets/1001890/d716e6d3-000c-47df-8369-fee66096d755">

My app was a fork from https://github.com/primefaces/sakai-react

### PrimeReact version

10.0.7

### React version

18.x

### Language

TypeScript

### Build / Runtime

Next.js

**Comment by melloware** [comment]
How long ago did you fork Sakai react?

Also is yours wrapped in PrimeReactProvider?

```xml
 <PrimeReactProvider>
                    <LayoutProvider>{children}</LayoutProvider>
</PrimeReactProvider>
```

**Comment by mondaychen** [comment]
@melloware not long ago. In fact, I can reproduce this issue with Sakai react now (master branch, latest commit https://github.com/primefaces/sakai-react/commit/8aa535b4a705b1c40678c81eb28748718672841b).

To reproduce

1. check out https://github.com/primefaces/sakai-react
2. upgrade primereact in package.json `"primereact": "^10.0.7"`
3. npm install
4. npm run dev

My node version is v20.5.0

Looks like `<PrimeReactProvider>` is already in https://github.com/primefaces/sakai-react/blob/8aa535b4a705b1c40678c81eb28748718672841b/app/layout.tsx#L21

**Comment by melloware** [comment]
Excellent I will check it out!

**Comment by melloware** [comment]
OK I fixed it correctly this time and tested with Sakai.  It will be fixed in 10.0.10
